bpf/lib/maps.h

```diff
@@ -99,6 +99,21 @@ struct {
 } AUTH_MAP __section_maps_btf;
 #endif
 
+#ifdef CONFIG_MAP
+/*
+ * CONFIG_MAP is an array containing runtime configuration information to the
+ * bpf datapath.  Each element in the array is 64 bytes, initially use of only
+ * one array element at index 0 is defined.
+ */
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__type(key, __u32);
+	__type(value, struct runtime_config_0);
```

Comment:
Using a struct here means that the smallest update you can make is the whole struct. In user space that will end up requiring a lookup of the current value, patching only the necessary parts and then writing that back. This is kind of unergonomic in terms of what code you write, and it also suffers badly if you have concurrent modifications.

What if instead of `struct runtime_config` we used `u64` here instead? That implies that the largest such runtime config is 8 bytes, but my gut feeling is that this is probably OK. To make "allocating" config easy you could add an enum.

Reply:
`u64` may be enough, but I was not ready to make that call without discussion. Current implementation caches the values in the userspace, so entries are updated only when needed. This design also periodically pulls in the config values from the configmap package so that the layout of the config data is only ever known in the configmap pkg and the datapath `lib/common.h` definitions. Also the size on 64 bytes likely makes no perf impact compared to the system call overhead. Also on the datapath side that is a typical cache line size, and I've also understood that bpf code accesses the map data by reference. Even if copied, a single cache line is likely not significantly slower than single `u64`.

That being sade, I had to choose a desing to propose. It would be nice if the datapath folks would discuss this over :-)